src/ncp-spinel/SpinelNCPControlInterface.cpp

```diff
@@ -309,6 +310,62 @@ SpinelNCPControlInterface::add_external_route(
 }
 
 void
+SpinelNCPControlInterface::joiner_add(
+		const uint8_t *addr,
+		int addr_len,
+		const char *psk,
+		int psk_len,
+		CallbackWithStatus cb
+) {
+
+	require_action(addr != NULL, bail, cb(kWPANTUNDStatus_InvalidArgument));
+	require_action(psk != NULL, bail, cb(kWPANTUNDStatus_InvalidArgument));
+	require_action(psk_len >= PSK_MIN_LENGTH, bail, cb(kWPANTUNDStatus_InvalidArgument));
+	require_action(psk_len <= PSK_MAX_LENGTH, bail, cb(kWPANTUNDStatus_InvalidArgument));
+	require_action(mNCPInstance->mEnabled, bail, cb(kWPANTUNDStatus_InvalidWhenDisabled));
+
+	mNCPInstance->start_new_task(SpinelNCPTaskSendCommand::Factory(mNCPInstance)
+		.set_callback(cb)
+		.add_command(SpinelPackData(
+			SPINEL_FRAME_PACK_CMD_PROP_VALUE_INSERT(
+			SPINEL_DATATYPE_EUI64_S
+			SPINEL_DATATYPE_DATA_S
+			SPINEL_DATATYPE_UINT8_S
+		),
+		SPINEL_PROP_THREAD_JOINERS,
+		addr,
+		psk,
+		psk_len
+	))
+	.set_lock_property(SPINEL_PROP_THREAD_ALLOW_LOCAL_NET_DATA_CHANGE)
+	.finish()
+	);
+
+bail:
+	return;
+}
+
+void
+SpinelNCPControlInterface::commissioner(
```

Comment:
Thank you, it looks way better. Changed to the wpan property. I think that support for commissioner --start / commissioner --stop commands should be present. Using wpanctl setprop / getprop and then commissioner --joiner-add does not seem to be very intuitive. Therefore --start / --stop which set spinel property are still there. Do you think such redundancy is acceptable in this case?